pkg/services/dashboards/models.go

```diff
@@ -60,6 +62,51 @@ func (d *Dashboard) SetVersion(version int) {
 	d.Data.Set("version", version)
 }
 
+func (d *Dashboard) ToResource() kinds.GrafanaResource[simplejson.Json, interface{}] {
+	res := kinds.GrafanaResource[simplejson.Json, interface{}]{
+		Kind:       "Dashboard",
+		APIVersion: "alphav1",
+		Metadata: kinds.GrafanaResourceMetadata{
+			Name:              d.UID,
+			Annotations:       make(map[string]string),
+			Labels:            make(map[string]string),
+			CreationTimestamp: v1.NewTime(d.Created),
+			ResourceVersion:   fmt.Sprintf("%d", d.Version),
+		},
+	}
+	if d.Data != nil {
+		copy := &simplejson.Json{}
+		db, _ := d.Data.ToDB()
+		_ = copy.FromDB(db)
+
+		copy.Del("id")
+		copy.Del("version") // ???
+		copy.Del("uid")     // duplicated to name
+		res.Spec = copy
+	}
+
+	res.Metadata.SetUpdatedTimestamp(&d.Updated)
+	if d.CreatedBy > 0 {
+		res.Metadata.SetCreatedBy(fmt.Sprintf("user:%d", d.CreatedBy))
+	}
+	if d.UpdatedBy > 0 {
+		res.Metadata.SetUpdatedBy(fmt.Sprintf("user:%d", d.UpdatedBy))
+	}
+	if d.PluginID != "" {
+		res.Metadata.SetOriginInfo(&kinds.ResourceOriginInfo{
+			Name: "plugin",
+			Key:  d.PluginID,
+		})
+	}
+	if d.FolderID > 0 {
+		res.Metadata.SetFolder(fmt.Sprintf("folder:%d", d.FolderID))
```

Comment:
This is instead of resolving FolderID into a uid?

Reply:
I'm not sure how to best resolve this -- maybe this is not the object to try to convert 🤔 

In real usage, i think we need a SQL query that includes the UID rather than a one-by-one lookup

Reply:
Yes, it'd be much more efficient to just join on the folder id and return the folder uid directly in the query vs having to do a separate query especially for lists.